src/app-layer-htp.c

```diff
@@ -2722,9 +2722,9 @@ static int HTPParserTest01(void)
     FAIL_IF(htp_tx_request_method_number(tx) != HTP_METHOD_POST);
     FAIL_IF(htp_tx_request_protocol_number(tx) != HTP_PROTOCOL_V1_0);
 
+    UTHFreeFlow(f);
     AppLayerParserThreadCtxFree(alp_tctx);
     StreamTcpFreeConfig(true);
```

Comment:
Flow worker seems to do the cleanup it in this order:
- Decode (leading to AppLayerParserThreadCtxFree)
- TCP
- Detect
- Output
- Flow

Could you please tell why do we choose to change the deinit order here instead of changing the init order?

Reply:
I'm a bit confused by what is being asked here. We're not in a full flow worker, not using threads. Just making sure that if a flow can have a reference to something global, we free the flow before the global thing.

Reply:
My expectation was that the unit test should resemble in a controlled way how we actually do things during runtime. Is it incorrect?

Reply:
It usually does the minimal thing.